app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -124,6 +124,7 @@ def new_vm(attr = { })
     def create_vm(args = { })
       vm = new_vm(args)
       create_volumes :prefix => vm.name, :volumes => vm.volumes, :backing_id => args[:image_id]
+      vm.iso_dir = vm.volumes.first.path.split('/')[0..-2].join('/')
```

Comment:
Actually, I think I take back point one.  It looks like the cloud-init disk that fog-libvirt generates is always going to be uploaded into the pool called "default" (due to upload_image).

If so, then there's probably no direct relation between the pool of the first volume and the ISO image's pool, and this code should be setting iso_dir to the `default` pool's path?  (This rather looks like a bug in fog-libvirt to me, not Foreman.)